### ISSUE TITLE
cite referencing varies with wiki

### ISSUE THREAD

**Issue by desb42** [gnosygnu/xowa#304]
Looking at the reference section of wikipedia eg 
![ref1](https://user-images.githubusercontent.com/861631/50267197-50a48500-041e-11e9-9a6b-0f6c567d9d1c.jpg)
Compared to xowa the character in the red oval is bold and the characters in the blue oval are both bold and italic
This is due to the fact that the templates for these are held in Mediawiki:Cite_references_link_one and Mediawiki:Cite_references_link_many respectively.
If these (and other related Cite_refs) are not defined for a wiki the fallback is to en.json (en.gfs in xowa)

For example
![ref2](https://user-images.githubusercontent.com/861631/50267617-c2310300-041f-11e9-97e3-b47f118cd25f.jpg)
Note the up arrow (this comes from the default en.json)

Ref_html_wtr_cfg.java contains fixed definitions for references regardless of wiki.
It seems that this ought to be using the Xow_msg_mgr style of messages.
This is a level of change that is currently beyond me.

**Comment by gnosygnu** [comment]
Yeah, references was one of the first extensions, and I hard-coded a lot of behavior in the reference classes. I'll make them more configuration-aware, but off the top of my head, this is sort of a medium task (4 hour-ish)

**Comment by desb42** [comment]
Having played with frwiki, the cite ref is quit different (no visible square brackets)
I was having a look at what could be done to achieve this.

fr.wikipedia.org/wiki/MediaWiki:Cite_reference_link contains

    <sup id="$1" class="reference">[[#$2|<span class="cite_crochet"><nowiki>[</nowiki></span>$3<span class="cite_crochet"><nowiki>]</nowiki></span>]]</sup>

I was playing with some logic that calls wiki.Msg_mgr().Val_by_key_args and saw that for a particular message a flag is set if there is any wikitext to interpret.

First of all (as far as I can tell) for this message the flag is not set (as I think the check is looking for curly braces), so the first attempt produced this

    <sup id="cite_ref-0" class="reference">[[#cite_note-0|<span class="cite_crochet"><nowiki>[</nowiki></span>note 1<span class="cite_crochet"><nowiki>]</nowiki></span>]]</sup>
If I force a template check I get

    <sup id="cite_ref-0" class="reference">[[#cite_note-0|<span class="cite_crochet">&xowa_brack_bgn;</span>note 1<span class="cite_crochet">&xowa_brack_end;</span>]]</sup>
closer(?)
It seems that a further pass (to convert lnki to html) is required

Although I have not checked, I would have expected this in other Mediawiki messages
